### PR TITLE
#3 song page details section

### PR DESCRIPTION
**Pull request by Toni-Zoaretz**

### CODE REVIEW

File: client/src/App.jsx [hunk 1]
```diff
@@ -2,13 +2,15 @@ import { useState } from "react";
 import reactLogo from "./assets/react.svg";
 import viteLogo from "/vite.svg";
 import "./App.css";
+import SongDetails from "./components/songDetails/SongDetails";
 
 function App() {
   const [count, setCount] = useState(0);
```
Comment (yishain11) [inline client/src/App.jsx:8]: remove vite boilerplate

File: client/src/App.jsx [hunk 2]
```diff
@@ -2,13 +2,15 @@ import { useState } from "react";
...
 
   return (
     <>
       <div>
+    <SongDetails/>
         <a href="https://vitejs.dev" target="_blank">
```
Comment (yishain11) [inline client/src/App.jsx:14]: remove vite boilerplate

File: client/src/App.jsx [hunk 3]
```diff
@@ -2,13 +2,15 @@ import { useState } from "react";
...
           <img src={viteLogo} className="logo" alt="Vite logo" />
```
Comment (yishain11) [inline client/src/App.jsx:15]: remove vite boilerplate

File: client/src/components/songDetails/Image.jsx [hunk 4]
```diff
@@ -0,0 +1,11 @@
+import styled from 'styled-components';
+
+export const Image = styled.img`
+  border-radius: 35px;
+  width:10rem;
+  height:10rem;
+`
+
```
Comment (yishain11) [inline client/src/components/songDetails/Image.jsx:8]: remove empty lines

File: client/src/components/songDetails/songDetails.css [hunk 5]
```diff
@@ -0,0 +1,45 @@
+
+  @import url('https://fonts.googleapis.com/css2?family=ABeeZee:ital@0;1&display=swap');
```
Comment (yishain11) [inline client/src/components/songDetails/songDetails.css:2]: better to download the font and put in assets, expecially in mobile first apps (where internet connection can be slow)

File: client/src/components/songDetails/Image.jsx [hunk 6]
```diff
@@ -0,0 +1,11 @@
+import styled from 'styled-components';
+
+export const Image = styled.img`
+  border-radius: 35px;
```
Comment (yishain11) [inline client/src/components/songDetails/Image.jsx:4]: remove non relative units

File: client/src/components/songDetails/songDetails.css [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+
```
Comment (yishain11) [inline client/src/components/songDetails/songDetails.css:3]: did you add mobile first? media queries? remember that this design works only for mobile screens - not tablet/laptop/pc

**Review by yishain11 — CHANGES_REQUESTED** [review]

File: client/package.json [hunk 8]
```diff
@@ -21,6 +21,7 @@
     "eslint-plugin-react": "^7.32.2",
     "eslint-plugin-react-hooks": "^4.6.0",
     "eslint-plugin-react-refresh": "^0.3.4",
+    "styled-components": "^5.3.10",
```
Comment (yishain11) [inline client/package.json:24]: styled component is not dev dependancy. its a regular dep

File: client/src/components/songDetails/songDetails.css [hunk 9]
```diff
@@ -0,0 +1,48 @@
+@font-face {
+  font-family: "ABeeZee";
+  src: url("../../assets/fonts/ABeeZee-Regular.ttf") format("truetype");
+}
+
+* {
```
Comment (yishain11) [inline client/src/components/songDetails/songDetails.css:6]: move this to main.css/app.css. this is relevant to all the application

**Review by yishain11 — CHANGES_REQUESTED** [review]
song details looks ok, but there are diffs from the figma style:
1. too much to the left, expecialy if you play with screen size (I tried galaxy S8+)
2. the circle aroung the letters are oval and not round + too close to the image. make sure that the spaces are the same between the elements.
3. artist name and song name are italic. click on figma design to make sure you dont miss styles + they are not in the middle like in figma
4. the vertical position is not right - should be @ the top of the page

File: client/src/components/songDetails/songDetails.css (hunk 9 above)
Comment (Toni-Zoaretz) [reply client/src/components/songDetails/songDetails.css:6]: do you mean to put it in my CSS file and **also** in the App.css file? or just in the App.css file in the root element?

File: client/src/components/songDetails/songDetails.css [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+    align-items: center;
+    min-width: 100%;
+    gap: 1.5rem;
+  }
+
+  .textDetailsContainer {
+    font-family: "ABeeZee Italic";
+    display: flex;
+    flex-direction: column;
+    gap: 7px;
```
Comment (yishain11) [inline client/src/components/songDetails/songDetails.css:19]: don't use none relative units

File: client/src/components/songDetails/songDetails.css [hunk 11]
```diff
@@ -0,0 +1,40 @@
...
+    margin-right: 18px;
```
Comment (yishain11) [inline client/src/components/songDetails/songDetails.css:20]: don't use none relative units

File: client/src/components/songDetails/songDetails.css [hunk 12]
```diff
@@ -0,0 +1,40 @@
...
+  .translation-div {
+    display: flex;
+    justify-content: center;
+    align-items: center;
+  }
+
+  .btn {
+    background-color: #d9d9d9;
+    border-radius: 100%;
+    padding: 10px 15px;
```
Comment (yishain11) [inline client/src/components/songDetails/songDetails.css:38]: don't use none relative units

File: client/src/components/songDetails/songDetails.css [hunk 13]
```diff
@@ -0,0 +1,40 @@
+@font-face {
+  font-family: "ABeeZee Italic";
+  src: url("../../assets/fonts/ABeeZee-Italic.ttf") format("truetype");
+}
+
+@media (min-width: 350px) and (max-width: 700px) {
+  .songDetailsContainer {
```
Comment (yishain11) [inline client/src/components/songDetails/songDetails.css:7]: for long container components, use the same style just as styled components

**Review by yishain11 — CHANGES_REQUESTED** [review]

**Pull request merged**